src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/FindPal.cs

```diff
@@ -0,0 +1,268 @@
+           
+            return results;
+        }
+
+        private static T ConfirmedCast<T>(object findValue)
+        {
+            Debug.Assert(findValue != null);
+
+            if (findValue.GetType() != typeof(T))
+                throw new CryptographicException(SR.Cryptography_X509_InvalidFindValue);
+
+            return (T)findValue;
+        }
+
+        private static string ConfirmedOidValue(IFindPal findPal, object findValue, OidGroup oidGroup)
+        {
+            string maybeOid = ConfirmedCast<string>(findValue);
+
+            if (maybeOid.Length == 0)
+            {
+                throw new ArgumentException(SR.Argument_InvalidOidValue);
+            }
+
+            return findPal.NormalizeOid(maybeOid, oidGroup);
+        }
+
+        private static X509KeyUsageFlags ConfirmedX509KeyUsage(object findValue)
+        {
+            if (findValue is X509KeyUsageFlags)
+                return (X509KeyUsageFlags)findValue;
+
+            if (findValue is int)
+                return (X509KeyUsageFlags)(int)findValue;
+
+            if (findValue is uint)
+                return (X509KeyUsageFlags)(uint)findValue;
+
+            string findValueString = findValue as string;
+
+            if (findValueString != null)
+            {
+                X509KeyUsageFlags usageFlags;
+
+                if (s_keyUsages.TryGetValue(findValueString, out usageFlags))
+                {
+                    return usageFlags;
+                }
+            }
+
+            throw new CryptographicException(SR.Cryptography_X509_InvalidFindValue);
+        }
+
+        //
+        // verify the passed keyValue is valid as per X.208
+        //
+        // The first number must be 0, 1 or 2.
+        // Enforce all characters are digits and dots.
+        // Enforce that no dot starts or ends the Oid, and disallow double dots.
+        // Enforce there is at least one dot separator.
+        //
+        internal static void ValidateOidValue(string keyValue)
+        {
+            if (keyValue == null)
+                throw new ArgumentNullException("keyValue");
+
+            int len = keyValue.Length;
+            if (len < 2)
+                throw new ArgumentException(SR.Argument_InvalidOidValue);
+
+            // should not start with a dot. The first digit must be 0, 1 or 2.
+            char c = keyValue[0];
+            if (c != '0' && c != '1' && c != '2')
+                throw new ArgumentException(SR.Argument_InvalidOidValue);
+            if (keyValue[1] != '.' || keyValue[len - 1] == '.') // should not end in a dot
+                throw new ArgumentException(SR.Argument_InvalidOidValue);
+
+            // While characters 0 and 1 were both validated, start at character 1 to validate
+            // that there aren't two dots in a row.
+            for (int i = 1; i < len; i++)
+            {
+                // ensure every character is either a digit or a dot
+                if (char.IsDigit(keyValue[i]))
```

Comment:
\u0662 (arabic 2) doesn't work instead of '2' on Windows, or Unix, returning the empty set instead of complaining the OID is invalid.  Adding a compatibility test, and just leaving this as matching Desktop's behavior.